packages/trader/src/AppV2/Components/TradeParameters/Stake/stake-input.tsx

```diff
@@ -0,0 +1,330 @@
+    amount?: string | number;
+    payout_per_point?: string | number;
+    barrier_1?: string | number;
+};
+
+const StakeInput = observer(({ onClose, is_open }: TStakeInput) => {
+    const trade_store = useTraderStore();
+    const {
+        amount,
+        commission,
+        contract_type,
+        currency,
+        barrier_1,
+        has_stop_loss,
+        is_accumulator,
+        is_multiplier,
+        is_turbos,
+        is_vanilla,
+        onChange,
+        proposal_info,
+        stop_out,
+        trade_type_tab,
+        trade_types,
+        validation_params,
+    } = trade_store;
+
+    const [proposal_request_values, setNewValues] = React.useState<TNewValues>({ amount }); // contains information for creating proposal request: stake (amount), payout_per_point for Turbos and barrier_1 for Vanillas
+    const [stake_error, setStakeError] = React.useState('');
+    const [fe_stake_error, setFEStakeError] = React.useState('');
+
+    const contract_types = getDisplayedContractTypes(trade_types, contract_type, trade_type_tab);
+    const should_show_payout_details = !is_accumulator && !is_multiplier && !is_turbos && !is_vanilla;
+
+    const {
+        contract_payout: first_contract_payout,
+        max_payout,
+        error: first_payout_error,
+    } = getPayoutInfo(proposal_info[contract_types[0]]);
+    const { contract_payout: second_contract_payout, error: second_payout_error } = getPayoutInfo(
+        proposal_info[contract_types[1]]
+    );
+
+    const { stake } = (validation_params[contract_types[0]] || validation_params[contract_types[1]]) ?? {};
+    const { max: max_stake = 0, min: min_stake = 0 } = stake ?? {};
+
+    const [details, setDetails] = React.useState({
+        commission,
+        error_1: first_payout_error,
+        error_2: second_payout_error,
+        first_contract_payout,
+        is_first_payout_exceed: !!first_payout_error && first_contract_payout > max_payout,
+        is_second_payout_exceed: !!second_payout_error && second_contract_payout > max_payout,
+        second_contract_payout,
+        max_payout,
+        max_stake,
+        min_stake,
+        stop_out,
+    });
+
+    // Parallel proposal without subscription
+    // For Rise/Fall and all Digits we should do 2 proposal requests
+    const should_send_multiple_proposals = contract_types.length > 1 && !is_multiplier;
+    const has_both_errors = !!details.error_1 && !!details.error_2;
+    // Need for cases with Rise/Fall and Digits, when only one response contains error and we should allow to save the value
+    const should_show_stake_error =
+        !should_send_multiple_proposals || (should_send_multiple_proposals && has_both_errors);
+
+    const proposal_req_1 = getProposalRequestObject({
+        new_values: proposal_request_values,
+        trade_store,
+        trade_type: contract_types[0],
+    });
+    const { data: response_1, is_fetching: is_fetching_1 } = useDtraderQuery<Parameters<TOnProposalResponse>[0]>(
```

Comment:
Instead of executing onChange function from trade store (that will change store values and send new proposal with them) when user is typing, we are using useDtraderQuery hook without subscription. It will allow us to validate the new values + we can extract more information from response. Because we are not subscribing, the store values won't be changed and we won't form a new proposal request with subscription. That will also allow us to restore original values in case of page reload & clicking on back arrow & closing without saving, so there is no need in stake in v2_params_initial_values. Only when user clicks on 'Save' button we'll call onChange.